### PR TITLE
O2-738 Bugfix: Sorting Tests CMake

### PR DESCRIPTION
**Pull request by MichaelLettrich**
Fixes bug that prevented sorting tests from being built because of wrong checks for a variable.

### CODE REVIEW

**Review by davidrohr — DISMISSED** [review]

**Comment by ktf** [comment]
Looks like you forgot a BOOST_TESTS_MAIN or something like that.

File: GPU/Common/CMakeLists.txt [hunk 1]
```diff
@@ -44,7 +44,7 @@ if(ALIGPU_BUILD_TYPE STREQUAL "O2")
                              GPUCA_TPC_GEOMETRY_O2 HAVE_O2HEADERS)
   
   # cuda test, only compile if CUDA
-  if(CUDA_ENABLED AND NOT $ENV{ALIBUILT_O2_TESTS})
+  if(ENABLE_CUDA AND NOT ENV{ALIBUILD_O2_TESTS})
```
Comment (MichaelLettrich) [inline GPU/Common/CMakeLists.txt]: after  discussion with @ktf we decided that this is not the right approach. Actually, we want to build the tests if we have CUDA and HIP available but exclude them from running when no cards are present, which can be done in the O2 recipe.

File: GPU/GPUTracking/Base/hip/CMakeLists.txt [hunk 2]
```diff
@@ -51,7 +51,7 @@ if(ALIGPU_BUILD_TYPE STREQUAL "O2")
 
   install(FILES ${HDRS} DESTINATION include/GPU)
 
- if(NOT $ENV{ALIBUILT_O2_TESTS})
+ if(ENABLE_HIP AND NOT ENV{ALIBUILD_O2_TESTS})
```
Comment (MichaelLettrich) [inline GPU/GPUTracking/Base/hip/CMakeLists.txt]: after  discussion with @ktf we decided that this is not the right approach. Actually, we want to build the tests if we have CUDA and HIP available but exclude them from running when no cards are present, which can be done in the O2 recipe.

Comment (davidrohr) [reply GPU/GPUTracking/Base/hip/CMakeLists.txt]: makes sense for me

**Comment by davidrohr** [comment]
Now we are hitting the gpu ci container problem.
This will only be fixed once we upgrade the GPU CI container, and add --always-prefer-system build flag. I am sorry, but I didn't have time yet to look in the alice aurora business. Will do so as soon as I find time.

**Comment by davidrohr** [comment]
@MichaelLettrich : Can you rebase and force push, CI should hopefully become green now

**Comment by davidrohr** [comment]
@MichaelLettrich : The CI passed this now, but we cannot merge because of the conflicts. Can you please rebase?

**Review by davidrohr — APPROVED** [review]

**Pull request merged**